software/NRG_itho_wifi/src/task_web.cpp

```diff
@@ -587,6 +587,124 @@ void handleAPI(AsyncWebServerRequest *request)
         return;
       }
     }
+    else if (strcmp(p->name().c_str(), "getsetting") == 0)
+    {
+      uint8_t idx = static_cast<uint8_t>(p->value().toInt());
+
+      if (ithoSettingsArray == nullptr)
+      {
+        request->send(400, "text/html", "The Itho settings array is NULL");
```

Comment:
Per [RFC 7231](https://www.rfc-editor.org/rfc/rfc7231#section-6.5.1) HTTP 400 seems the most reasonable status, instead of 422 (which I think is rarely used anyway). I opted not to use HTTP 200 with a "NOK" response as that doesn't say much about why the request failed.

Reply:
I don't agree about the statement about `422` but I do agree, that this indeed is supposed to be classified as `400`

Reply:
400 is fine, 422 maybe even better. Both are better than NOK. NOK should have been replaced a lot earlier. NOK is a remnant of a years old idea (actually another project even) about a very simple web based / url encoded api thingy.... as how these things go, especially if you lack any experience thinking these things through towards the future. 
I now at least understand better how legacy comes about ;-)